test/snapshot-serializers/canvas-rendering-context.ts

```diff
@@ -0,0 +1,58 @@
+import { Printer, Config, Refs } from 'pretty-format';
+
+const isCanvasCtx = (val: unknown): val is CanvasRenderingContext2D =>
+  // native browser
+  (typeof CanvasRenderingContext2D !== 'undefined' &&
+    val instanceof CanvasRenderingContext2D) ||
+  // node‑canvas / jsdom polyfill: heuristics
+  !!(
+    val &&
+    typeof val === 'object' &&
+    typeof (val as any).drawImage === 'function' &&
+    (val as any).canvas?.tagName === 'CANVAS'
+  );
+
+/**
+ * Snapshot serializer that prints a stable, cross‑runtime representation
+ * of CanvasRenderingContext2D.
+ */
+export default {
+  test: isCanvasCtx,
+
+  serialize(val: any, config: Config, indentation: string, depth: number, refs: Refs, printer: Printer) {
```

Comment:
this is the serializer I mentioned earlier to have stable snapshot cross env for canvas rendering context

Reply:
For me this can return the type only as a string. 'CanvasRenderingContext2d` this because in future this block of code may become a sleeping issue for someone else approaching unit test. We don't need to verify the CTX below the surface of being a CTX, so this may be unnecessary. Sorry if you spent a bunch of time on this

Reply:
> For me this can return the type only as a string. 'CanvasRenderingContext2d` this because in future this block of code may become a sleeping issue for someone else approaching unit test. We don't need to verify the CTX below the surface of being a CTX, so this may be unnecessary. Sorry if you spent a bunch of time on this

don't worry
I also learned something new along the way
I will adjust the serializer to return the type only, it makes sense to me as well